lib/ain-ocean/src/api/rawtx.rs

```diff
@@ -1,27 +1,148 @@
+        }
+        Err(e) => {
+            eprintln!("Failed to send raw transaction: {:?}", e);
+            Err(Error::RpcError(e))
+        }
+    }
 }
 
-async fn test_rawtx() -> String {
-    "Testing raw transaction".to_string()
+#[ocean_endpoint]
+async fn get_raw_tx(
+    Path((txid, verbose)): Path<(String, bool)>,
+    Extension(ctx): Extension<Arc<AppContext>>,
+) -> Result<Response<RawTransaction>> {
+    let tx_hash = Txid::from_str(&txid)?;
+    let tx_result = ctx.client.get_transaction(&tx_hash, Some(verbose)).await?;
+    let details: Vec<TransctionDetails> = tx_result
+        .details
+        .into_iter()
+        .map(|detail| TransctionDetails {
+            address: detail.address,
+            category: detail.category.to_owned(),
+            amount: detail.amount.to_sat(),
+            label: detail.label,
+            vout: detail.vout,
+            fee: detail.fee.map(|f| f.to_sat()),
+            abandoned: detail.abandoned,
+            hex: hex::encode(&tx_result.hex),
+            blockhash: tx_result.info.blockhash,
+            confirmations: tx_result.info.confirmations,
+            time: tx_result.info.time,
+            blocktime: tx_result.info.blocktime,
+        })
+        .collect();
+
+    let bip_125 = match tx_result.info.bip125_replaceable {
+        Bip125Replaceable::Yes => Some("Yes".to_string()),
+        Bip125Replaceable::No => Some("No".to_string()),
+        Bip125Replaceable::Unknown => Some("Unknown".to_string()),
+    };
+
+    let raw_tx = RawTransaction {
+        info: WalletTxInfo {
+            confirmations: tx_result.info.confirmations,
+            blockhash: tx_result.info.blockhash,
+            blockindex: tx_result.info.blockindex,
+            blocktime: tx_result.info.blocktime,
+            blockheight: tx_result.info.blockheight,
+            txid: tx_result.info.txid,
+            time: tx_result.info.time,
+            timereceived: tx_result.info.timereceived,
+            bip125_replaceable: bip_125,
+            wallet_conflicts: tx_result.info.wallet_conflicts,
+        },
+        amount: tx_result.amount.to_sat(),
+        fee: tx_result.fee.map(|amount| amount.to_sat()),
+        details,
+        hex: tx_result.hex,
+    };
+    Ok(Response::new(raw_tx))
 }
 
-async fn get_rawtx(Path(txid): Path<String>) -> String {
-    format!("Details of raw transaction with txid {}", txid)
+fn validate(hex: String) -> Result<()> {
+    println!("{:?}", hex);
+    if !hex.starts_with("040000000001") {
+        return Err(Error::ValidationError(
+            "Transaction does not start with the expected prefix.".to_string(),
+        ));
+    }
+    Ok(())
 }
 
 pub fn router(ctx: Arc<AppContext>) -> Router {
+    println!("{:?}", ctx.network);
     Router::new()
         .route("/send", post(send_rawtx))
-        .route("/test", get(test_rawtx))
-        .route("/:txid", get(get_rawtx))
+        .route("/test", post(test_rawtx))
+        .route("/:txid", get(get_raw_tx))
```

Comment:
hmm.. can standardize the naming??
```suggestion
        .route("/test", post(test_rawtx))
        .route("/:txid", get(get_raw_tx))
```

Reply:
Its same as in `jf`

Reply:
i mean the function naming either we go for `test_rawtx` or `test_raw_tx`.. we need to standardize the naming

Reply:
its not fixed